packages/medusa/src/repositories/product-category.ts

```diff
@@ -69,17 +78,34 @@ export const ProductCategoryRepository = dataSource
         delete options_.where?.name
         delete options_.where?.handle
 
-        options_.where = [{
-          ...options_.where,
-          name: ILike(`%${q}%`)
-        }, {
-          ...options_.where,
-          handle: ILike(`%${q}%`)
-        }]
+        options_.where = [
+          {
+            ...options_.where,
+            name: ILike(`%${q}%`),
+          },
+          {
+            ...options_.where,
+            handle: ILike(`%${q}%`),
+          },
+        ]
       }
 
       queryBuilder.where(options_.where)
 
+      if (Array.isArray(depth) && depth.length > 0) {
+        // Depth is a number greater than 0, with this regex, we are checking how
+        // many period (.) characters are present in a record. A top level category
+        // has one period, this is depth 1. Depth = total of all periods in mpath string
+        const regexPatterns = depth
```

Comment:
would make sense to store the depth level of each category in the DB as a new column so it can be use for filtering in general? That sounds like a more performant and reliable approach

Reply:
Nice, good point. I have an env with 4k categories. I'll run some benchmarks on it. If it doesn't look too bad by coupling this with some other heavy queries (include descendants + search), I'll keep it as is for now, if not I'll create a new column to store the depth.